src/matrix/kernelkroneckermat.jl

```diff
@@ -27,31 +27,42 @@ end
 """
 @inline iskroncompatible(κ::Kernel) = false # Default return for kernels
 
-function _kernelmatrix_kroneckerjl_helper(::MOInputIsotopicByFeatures, Kfeatures, Koutputs)
+function _kernelmatrix_kroneckerjl_helper(
+    ::Type{<:MOInputIsotopicByFeatures}, Kfeatures, Koutputs
+)
     return Kronecker.kronecker(Kfeatures, Koutputs)
 end
 
-function _kernelmatrix_kroneckerjl_helper(::MOInputIsotopicByOutputs, Kfeatures, Koutputs)
+function _kernelmatrix_kroneckerjl_helper(
+    ::Type{<:MOInputIsotopicByOutputs}, Kfeatures, Koutputs
+)
     return Kronecker.kronecker(Koutputs, Kfeatures)
 end
 
+"""
+    kronecker_kernelmatrix(
+        k::Union{IndependentMOKernel,IntrinsicCoregionMOKernel}, x::MOI, y::MOI
+    ) where {MOI<:IsotopicMOInputsUnion}
+
+Requires Kronecker.jl: Computes the `kernelmatrix` for the `IndependentMOKernel` and the
+`IntrinsicCoregionMOKernel`, but returns a lazy kronecker product. This object can be very
+efficiently inverted or decomposed. See also [`kernelmatrix`](@ref).
+"""
 function kronecker_kernelmatrix(
-    k::Union{IndependentMOKernel,IntrinsicCoregionMOKernel},
-    x::IsotopicMOInputsUnion,
-    y::IsotopicMOInputsUnion,
-)
+    k::Union{IndependentMOKernel,IntrinsicCoregionMOKernel}, x::MOI, y::MOI
+) where {MOI<:IsotopicMOInputsUnion}
     @assert x.out_dim == y.out_dim
```

Comment:
as pointed out to me on my own PR, this should probably be an explicit error:
```suggestion
    x.out_dim == y.out_dim || throw(DimensionMismatch("`x` and `y` must have the same `out_dim`"))
```